packages/syrup/src/symbol.js

```diff
@@ -0,0 +1,3 @@
+// To be used as keys, syrup symbols must be javascript symbols.
+// To avoid an otherwise meaningful symbol name, we prefix it with 'syrup:'.
+export const SyrupSymbolFor = (name) => Symbol.for(`syrup:${name}`);
```

Comment:
This unfortunately registers every symbol encountered in a global table that isn’t necessarily garbage collected.

My intuition is that we’ll need to either wrap _every_ value in a wrapper object like `{type: 'symbol', symbol: 'hello'}` or just wrap all non-primitive values, including `list`, `struct`, and `record`, like `{type: 'record', record: [1, 2, 3]}` and `{type: 'list', list: [1, 2, 3]}`.

Reply:
This is also why we’re now calling Syrup symbols “Selector”. They do not necessarily correspond to implementation-language symbols.

Reply:
i made them prefixed symbols so that they can be keys on dictionaries (represented as js objects) though I don't create them when they are Record labels, as used in ocapn messages.

could create a custom clearable symbol registry though `Symbol('hi') !== Symbol.for('hi')`

as well as other options